README.md

```diff
@@ -116,16 +116,18 @@ In particular, when creating a document, we use the following in place of the cu
 1. If _key_ is an [origin](https://html.spec.whatwg.org/multipage/origin.html#concept-origin), then return _origin_.
 1. If _group_'s [agent cluster map](https://html.spec.whatwg.org/#agent-cluster-map)\[_origin_] exists, then return _origin_.
 1. If _requestsIsolation_ is true:
-    1. If _group_'s [agent cluster map](https://html.spec.whatwg.org/#agent-cluster-map)\[_site_] exists, and _group_'s [agent cluster map](https://html.spec.whatwg.org/#agent-cluster-map)\[_site_] contains any agents which contain any realms whose [settings object](https://html.spec.whatwg.org/#concept-realm-settings-object)'s [origin](https://html.spec.whatwg.org/#concept-settings-object-origin) are [same-origin](https://html.spec.whatwg.org/#same-origin) with _origin_, then return _site_.
+    1. If _group_'s [agent cluster map](https://html.spec.whatwg.org/#agent-cluster-map)\[_site_] exists, and _group_ has seen _origin_ before, then return _site_.
     1. Return _origin_.
 1. Return _site_.
 
+A browsing context group **has seen an origin before** if any of the browsing contexts in the group have at some point contained realms whose global object's associated `Document`'s origin matches the origin in question. (This will be formalized in the actual specification by tracking a list per BCG. Note that iframes that get removed from the document are no longer in the group, so do not contribute to this calculation.)
+
 This algorithm has two interesting features:
 
 * Even if origin isolation is not requested for a particular document creation, if origin isolation has previously created an origin-keyed agent cluster, then we put the new document in that origin-keyed agent cluster.
-* Even when origin isolation is requested, if there is a site-keyed agent cluster with same-origin documents, then we put the new document in that site-keyed agent cluster, ignoring the isolation request.
+* Even when origin isolation is requested, if there is a site-keyed agent cluster with same-origin documents in the browsing context group, then we put the new document in that site-keyed agent cluster, ignoring the isolation request.
 
-Both of these are consequences of a desire to ensure that same-origin sites do not end up isolated from each other.
+Both of these are consequences of a desire to ensure that same-origin sites do not end up isolated from each other, even in scenarios involving navigating through the session history.
```

Comment:
This seems tricky as this can involve multiple BCGs. https://github.com/whatwg/html/issues/5350 will be relevant.

Reply:
This sentence was meant to be more of a non-normative summary, and not a strict definition, but I can see how it's confusing in that way.